### DISCUSSION TITLE
Is it possible to set encoding type for form data of httpx.post?

### DISCUSSION THREAD

**Discussion by Nisekoi5** [encode/httpx#2786] [Q&A] [answered]
I use machine translation

Some websites do not use utf-8 (They are too old 😀) , and using utf-8 encoding will cause it to fail to work 
Is there a way to explicitly specify the encoding? My manual transmission of encoded data will result in a second encoding
I tried the following methods and it doesn't work
```
data = {
    "message1": "测试",
    "message2": "测试".encode("gbk"),
    "message3": quote("测试", encoding="gbk"),
}
result = httpx.post("https://httpbin.org/post", data=data)
# data obtained
  "form": {
    "message1": "\u6d4b\u8bd5",
    "message2": "b'\\xb2\\xe2\\xca\\xd4'",
    "message3": "%B2%E2%CA%D4"
  }
```
The way the form is coded seems to be written in the source code? Is there a way to modify the encoding when sending?
thanks for your answer
https://github.com/encode/httpx/blob/6a1841b924ab5d318d771e1a04ffb55662bf458b/httpx/_content.py#L145

**Comment by lovelydinosaur** [comment, marked as answer]
> Is there a way to modify the encoding when sending?

I guess you'd need to send this as pre-encoding bytes, setting the Content-Type header explicitly?...

```python
data = {"message1": "测试".encode("gbk")}
content = urlencode(data)
httpx.post(url, content=content, headers={"Content-Type": "application/x-www-form-urlencoded"})
```

**Reply by Nisekoi5** [reply]
This works fine, thanks for your answer!
